gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java

```diff
@@ -149,30 +151,29 @@ public class PlanServiceImpl extends AbstractService implements PlanService {
 
     @Override
     public PlanEntity findById(final ExecutionContext executionContext, final String plan) {
-        return (PlanEntity) planSearchService.findById(executionContext, plan);
+        return PlanAdapter.INSTANCE.map(planSearchService.findById(executionContext, plan));
     }
 
     @Override
     public Set<PlanEntity> findByApi(final ExecutionContext executionContext, final String api) {
-        return planSearchService.findByApi(executionContext, api).stream().map(PlanEntity.class::cast).collect(Collectors.toSet());
+        return stream(planSearchService.findByApi(executionContext, api)).flatMap(this::map).collect(Collectors.toSet());
     }
 
     @Override
     public PlanEntity create(final ExecutionContext executionContext, NewPlanEntity newPlan) {
         try {
             logger.debug("Create a new plan {} for API {}", newPlan.getName(), newPlan.getApi());
 
-            assertPlanSecurityIsAllowed(executionContext, newPlan.getSecurity());
-
             Api api = apiRepository.findById(newPlan.getApi()).orElseThrow(() -> new ApiNotFoundException(newPlan.getApi()));
+            assertPlanSecurityIsAllowed(executionContext, newPlan.getSecurity(), api.getDefinitionVersion());
 
             if (api.getApiLifecycleState() == DEPRECATED) {
-                throw new ApiDeprecatedException(api.getName());
+                throw new ApiDeprecatedException(api.getId());
```

Comment:
I understand the change but I thing it would have been better to isolate that in a dedicated commit because it has a functional impact